tests/scripts/test_generate_data.py

```diff
@@ -126,3 +133,82 @@ def test_generate_classification_data(temporary_dir):
+    output_header = os.path.join(temporary_dir, task_type, "header")
+
+    # create test arguments for the script
+    script_args = [
+        "generate.py",
+        "--train_samples", "100",
+        "--train_partitions", "4",
+        "--test_samples", "10",
+        "--test_partitions", "2",
+        "--inferencing_samples", "100",
+        "--inferencing_partitions", "1",
+        "--n_features", "40",
+        "--n_informative", "10",
+        "--random_state", "5",
+        "--output_train", output_train,
+        "--output_test", output_test,
+        "--output_inference", output_inference,
+        "--output_header", output_header,
+        "--type", task_type,
+    ]
+    if task_type == "classification":
+        script_args.extend(["--n_redundant", "5"])
+
+    if task_type == "lambdarank":
+        script_args.extend(["--docs_per_query", "5"])
+        script_args.extend(["--n_label_classes", "3"])
+
+    # replaces sys.argv with test arguments and run main
+    with patch.object(sys, "argv", script_args):
+        generate.main()
+
+    # test expected outputs
+    assert os.path.isfile(
+        os.path.join(output_train, "train_0.txt")
+    ), "Script generate.py should generate train_0.txt under --output dir but did not"
+    assert os.path.isfile(
+        os.path.join(output_train, "train_1.txt")
+    ), "Script generate.py should generate train_1.txt under --output dir but did not"
+    assert os.path.isfile(
+        os.path.join(output_train, "train_2.txt")
+    ), "Script generate.py should generate train_2.txt under --output dir but did not"
+    assert os.path.isfile(
+        os.path.join(output_train, "train_3.txt")
+    ), "Script generate.py should generate train_3.txt under --output dir but did not"
+    assert not os.path.isfile(
+        os.path.join(output_train, "train_4.txt")
+    ), "Script generate.py should NOT generate train_4.txt under --output dir but DID"
+
+    assert os.path.isfile(
+        os.path.join(output_test, "test_0.txt")
+    ), "Script generate.py should generate test_0.txt under --output dir but did not"
+    assert os.path.isfile(
+        os.path.join(output_test, "test_1.txt")
+    ), "Script generate.py should generate test_1.txt under --output dir but did not"
+    assert not os.path.isfile(
+        os.path.join(output_test, "test_2.txt")
+    ), "Script generate.py should NOT generate test_2.txt under --output dir but DID"
+
+    assert os.path.isfile(
+        os.path.join(output_inference, "inference_0.txt")
+    ), "Script generate.py should generate inference_0.txt under --output dir but did not"
+    assert not os.path.isfile(
+        os.path.join(output_inference, "inference_1.txt")
+    ), "Script generate.py should NOT generate inference_1.txt under --output dir but DID"
+
+    assert os.path.isfile(
+        os.path.join(output_header, "header.txt")
+    ), "Script generate.py should generate inference_0.txt under --output dir but did not"
```

Comment:
wrong comment here, should read "header.txt" instead of inference_0.txt

Reply:
done